lib/excon/utils.rb

```diff
@@ -87,7 +87,8 @@ def parse_query_string(string)
     end
 
     def default_port?(datum)
-      (datum[:scheme].casecmp('http').zero? && datum[:port] == 80) ||
+      datum[:port].nil? ||
+        (datum[:scheme].casecmp('http').zero? && datum[:port] == 80) ||
```

Comment:
Don't you still want to make it safe if no scheme is known?
```suggestion
        (datum[:scheme]&.casecmp('http')&.zero? && datum[:port] == 80) ||
```

I'm also surprised that here the scheme is case insensitive but in other places the `HTTPS` constant is used.

https://github.com/excon/excon/blob/a8f625d1bd5eee8d99371b38751ac097562dcf23/lib/excon/connection.rb#L554

The constant is defined here:
https://github.com/excon/excon/blob/a8f625d1bd5eee8d99371b38751ac097562dcf23/lib/excon/constants.rb#L30

Reply:
> Don't you still want to make it safe if no scheme is known?

Done

> I'm also surprised that here the scheme is case insensitive but in other places the HTTPS constant is used.

It is odd, but it has been this way for over a decade, so I'd rather let it be

Reply:
I think it might be one of those "be generous with input, be strict with output" kind of things in terms of capitalization. But it's been so long I don't recall for sure. It might be worth thinking about more and cleaning up further, but this seems good-enough for now.